internal/scheduler/validation.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2025 SAP SE
+// SPDX-License-Identifier: Apache-2.0
+
+package scheduler
+
+import (
+	"errors"
+	"log/slog"
+
+	"github.com/cobaltcore-dev/cortex/internal/conf"
+	"github.com/cobaltcore-dev/cortex/internal/db"
+	"github.com/cobaltcore-dev/cortex/internal/scheduler/api"
+	"github.com/cobaltcore-dev/cortex/internal/scheduler/plugins"
+)
+
+// The config type has a long name, so we use a shorter alias here.
+// The name is intentionally long to make it explicit that we disable
+// validations for the scheduler step instead of enabling them.
+type disabledValidations = conf.SchedulerStepDisabledValidationsConfig
+
+// Wrapper for scheduler steps that validates them before/after execution.
+type StepValidator[S plugins.Step] struct {
+	// The wrapped step to validate.
+	Step S
```

Comment:
```suggestion
	Step plugins.Step
```
What does the generic bring as value?

Reply:
Actually none here. Will adjust this